### PR TITLE
[Quest] All At Sea

### PR DESCRIPTION
**Pull request by jamesbradleym**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Implements quest All At Sea, adapted from [ASB Commit](https://github.com/AirSkyBoat/AirSkyBoat/commit/2fa80b665a1f4ee379d4d6924ce49496dfed9dc4)

## Steps to test these changes

- Speak with Paytah in Port Windurst with >= 3 Windy fame (!gotoid 17760350)
- Trade Ripped Cap (!additem 591) to Paytah who cries
- Trade Ripped Cap to Baren-Moren (!gotoid 17752232)
- Trade 4x Dhalmel Hide to Baren-Moren (!additem 857 4)
- Trade in Sailor's Cap to Paytah who doesn't cry
- ??
- Profit

### CODE REVIEW

File: scripts/quests/windurst/All_At_Sea.lua [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(291)
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, xi.item.RIPPED_CAP) then
+                        return quest:progressEvent(292)
+                    end
+                end,
```
Comment (Xaver-DaRed) [inline scripts/quests/windurst/All_At_Sea.lua:38]: Can you really start the quest directly by trading without speaking to the NPC?

File: scripts/quests/windurst/All_At_Sea.lua [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, xi.item.SAILORS_CAP) and
+                        quest:getVar(player, 'Prog') == 4
+                    then
+                        return quest:progressEvent(295) -- Sailor's Cap turn in
+                    elseif quest:getVar(player, 'Prog') == 4 then
+                        return quest:progressEvent(298) -- Can't accept what isn't theirs
+                    end
```
Comment (Xaver-DaRed) [inline scripts/quests/windurst/All_At_Sea.lua:83]:
```lua
        if quest:getVar(player, 'Prog') == 4 then
            if npcUtil.tradeHasExactly(trade, xi.item.SAILORS_CAP) then
                return quest:progressEvent(295) -- Sailor's Cap turn in
            else
                return quest:progressEvent(298) -- Can't accept what isn't theirs
            end
        end
```

File: scripts/quests/windurst/All_At_Sea.lua (hunk 1 above)
Comment (jamesbradleym) [reply scripts/quests/windurst/All_At_Sea.lua:38]: Good point, speaking to Paytah does not initiate the quest but I've put in a capture request to confirm you don't need to speak with him at all

File: scripts/quests/windurst/All_At_Sea.lua (hunk 2 above)
Comment (jamesbradleym) [reply scripts/quests/windurst/All_At_Sea.lua:83]: Re-organized ♻️

File: scripts/quests/windurst/All_At_Sea.lua (hunk 1 above)
Comment (jamesbradleym) [reply scripts/quests/windurst/All_At_Sea.lua:38]: https://www.youtube.com/watch?v=NyrhHhkVydk

Confirmed by capture, you can indeed start the quest with a trade and without speaking to NPC

**Comment by jamesbradleym** [comment]
@Xaver-DaRed Heya, checking in on this :)

**Comment by jamesbradleym** [comment]
Closing this PR to reduce open PRs and prevent staleness—will revisit.

**Pull request closed without merging**